software-containers/README.md

```diff
@@ -16,3 +16,156 @@ group:      sss files
+
+For work that does not require a UI, for example adding a new package to CASA, we may be able to run the container and test our work at this point. Execute 'docker run -it <docker image ID> /bin/bash' and check if the work is correct.
+### Deploying A Container
+Since a CASA container runs in Arcade under Kubernetes, the container image generated from the build step needs to be pushed to the repository that Kubernetes uses.
+1. log into docker by entering your username and password(this only needs to be done once)
+```
+docker login bucket.canfar.net
+```
+2. push the docker image generate in the build step to bucket.canfar.net
+```
+docker push bucket.canfar.net/arcade-casa:<TAG>
+```
+3. ssh to canfar-login
+```
+ssh canfar-login
+```
+4. pull the image into canfar-login
+```
+docker pull bucket.canfar.net/arcade-casa:<TAG>
+```
+5. tag the image with the repository that Kubernetes uses
+```
+docker tag <docker image ID> canfar-registry.cloud.computecanada.ca/arcade-casa:<TAG>
+```
+6. push the image to the repository that Kubernetes uses
+```
+docker push canfar-registry.cloud.computecanada.ca/arcade-casa:<TAG>
+```
+### Testing A Container
+Since a CASA container resides inside Arcade, we need to access an Arcade desktop session. We can either access an existing Arcade session or we can create one.
+#### Get A Session
+##### Open An Existing Arcade Session
+Execute the following command and open a browser using the desktop session URL returned
+```
+curl -E $A/test-certificates/x509_CADCAuthtest2.pem https://proto.canfar.net/arcade/session
+```
+##### Create Your Own Desktop Session
+Execute the following command and open a browser using the desktop session URL returned
+```
+curl -E ~/.ssl/cadcproxy.pem https://proto.canfar.net/arcade/session -d "name=mydesktop" -d "type=desktop"
+```
+#### Get Access The CASA Container
+##### Use A Custom Script
+We can create a custom script to access the container. We can save the script in our home directory, which persists over sessions. To access the CASA container, we just execute the script. The following is an example of such a script. Just replace the version number (4.7.0-el7) with the version under test. An xterm for the CASA version will pop up after some time.
+```
+#!/bin/bash
+
+IP_ADDRESS=`hostname --ip-address`
+
+/opt/shibboleth/bin/curl -v -L -k -E /cavern/home/${arcade_username}/.ssl/cadcproxy.pem -d "target-ip=${IP_ADDRESS}" -d "software=canfar-registry.cloud.computecanada.ca/arcade-casa:4.7.0-el7" https://${arcade_hostname}/arcade/session/${VNC_PW}/app
+```
+##### Use The UI
+If you have previously created a desktop session, you can use the UI of that session to access your container. There is an 'Applications' tab on the top left hand corner. Select 'Applications->CANFAR->CASA' and then the version under development. An xterm for the selected CASA version will pop up after some time.
+#### Perform Tests
```

Comment:
This information in 'perform tests' is good so you can keep this.

Reply:
Okay